### PR TITLE
CORE-104: add support for nested AVUs.

### PR DESCRIPTION
**Pull request by slr71**
This is a work in progress. The preliminary coding and testing are done, but I still want to proofread the code before definitively saying that it's ready. I also need to add documentation. I wanted to submit this PR relatively soon, however, so that I can get some other eyes on the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,99 @@ user=> (println (indent-str xml))
 ;; output omitted for brevity.
 ```
 
+### DataCite 4.1
+
+The primary interface for creating a DataCite 4.1 file is essentially the same as the interface for creating a DataCite
+3.1 file. The function `org.cyverse.metadata-files.datacite-4-1/build-datacite` returns a an instance of
```
Comment (psarando) [inline README.md]:
```suggestion
3.1 file. The function `org.cyverse.metadata-files.datacite-4-1/build-datacite` returns an instance of
```

File: src/org/cyverse/metadata_files/datacite_4_1/geo_locations.clj [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+(alias-uris)
+
+;; General longitude elements
+
+(defn- validate-longitude [location longitude]
+  (try
+    (let [longitude (Double/parseDouble longitude)]
+      (when-not (<= -180.0 longitude 180.0)
+        (throw (ex-info "Longitudes must be between -180.0 and 180.0" {:location location :longitude longitude}))))
```
Comment (psarando) [inline src/org/cyverse/metadata_files/datacite_4_1/geo_locations.clj:14]: :clojure: makes this range check so easy 👍

File: src/org/cyverse/metadata_files/util.clj [hunk 3]
```diff
@@ -54,3 +59,76 @@
         ovs        (extend-associated-optional-values rvs ovs)]
     (validate-associated-required-values required-attribute-names optional-attribute-names rvs ovs)
     (drop-while (partial every? string/blank?) (apply map vector (concat rvs ovs)))))
+
+(defn- get-attr-counts [attributes]
+  (reduce #(update %1 %2 (fnil inc 0)) {} (map :attr attributes)))
```
Comment (psarando) [inline src/org/cyverse/metadata_files/util.clj:64]: 👍

**Review by psarando — APPROVED** [review]
LGTM 👍  I didn't catch any issues, though this turned out to be a lot more work than I was expecting 😰 

The `new-simple-nested-element-generator` and `new-container-nested-element-generator` functions make the rest of this code very easy to follow ✨ 
 :shipit:

File: src/org/cyverse/metadata_files/datacite_4_1/geo_locations.clj (hunk 2 above)
Comment (slr71) [reply src/org/cyverse/metadata_files/datacite_4_1/geo_locations.clj:14]: Yeah, it really does.

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**